linkcheck/checker/httpurl.py

```diff
@@ -273,7 +272,7 @@ def follow_redirections(self, request):
             # Reset extern and recalculate
             self.extern = None
             self.set_extern(newurl)
-            self.urlparts = strformat.url_unicode_split(newurl)
+            self.urlparts = list(urllib.parse.urlsplit(newurl))
```

Comment:
Life would be better if we could retain the named tuple, but IIRC there's code that wants self.urlparts to be a mutable list.